web/client/components/share/SharePanel.jsx

```diff
@@ -97,27 +105,51 @@ class SharePanel extends React.Component {
             embed: 3
         };
         const bbox = join(this.props.bbox, ',');
+        const coordinate = this.getCoordinates(this.props);
         this.setState({
             bbox,
-            eventKey: tabs[this.props.selectedTab] || 1
+            eventKey: tabs[this.props.selectedTab] || 1,
+            zoom: this.props.zoom,
+            coordinate
         });
     }
 
     UNSAFE_componentWillReceiveProps(newProps) {
-        const bbox = join(this.props.bbox, ',');
         const newBbox = join(newProps.bbox, ',');
-        if (bbox !== newBbox) {
+        if (this.props !== newProps) {
```

Comment:
is better to use isEqual from lodash to compare objects.
but i would sugges to to add a list of condition in OR with each other.
this needs to be done only for the props you need to listen for change